### ISSUE TITLE
Update Nukkit to support client verstion 1.1.0

### ISSUE THREAD

**Issue by dcmwai** [Nukkit/Nukkit#1743]
Unable to join server when client version is 1.10 (Win 10 PE/iOS

---
Edit by @Snake1999:

**We need your help for testing our software for client 1.1.0!** Compile code on https://github.com/yescallop/Nukkit/tree/1.1.0 and try it out~ After tested OK, our main branch will be updated to 1.1.0. 

We are now working 1.1.0 on: https://github.com/yescallop/Nukkit/tree/1.1.0
Any contributions please go to the site above. We welcome every bit of your work.

Go on nukkiting out our code :)

---

As our new branch is bleeding - still under construction and maybe buggy or unfit for production use, no jar file are available for 1.1.0 yet unless you do the compile work by yourself. @PetteriM1 

---

**Comment by yungtechboy1** [comment]
👍

**Comment by imjack** [comment]
That's because it's not updated to the latest version

**Comment by MasterlazorX** [comment]
It's not updated to to 1.1 yet.

**Comment by kvetinac97** [comment]
Anyways, somebody should update it. @yescallop ?

**Comment by MasterlazorX** [comment]
I'm sure there's a BUNCH of protocol changes to mcpe 1.1.

**Comment by Creeperface01** [comment]
or @kvetinac97 ?

**Comment by jrodr10** [comment]
We're gonna need a binary rewire & nbt rewire & protocol change to 113 ._.

**Comment by DaMatrix** [comment]
...and this is where nukkit dies. with no active devs and a massive update like this, we ain't gonna be getting an update at this rate. :/

**Comment by RednedEpic** [comment]
1.1 came out YESTERDAY. I bet the developers are hard at work at trying their best to update Nukkit to 1.1. As a developer, once there is a new update of Minecraft to update plugins to. It definitely takes a lot longer than a day to update. And if you don't believe me, go ahead, Fork Nukkit and update it yourself.

**Comment by DaMatrix** [comment]
i know more than a bit about what's required, i'm making a general commentary that the original devs are pretty much dead and nukkit doesn't get much any more other than occastional version bumps and a PR every once in a while.

**Comment by boybook** [comment]
I'm trying to write about it but lots of problems.

**Comment by ScraM-Team** [comment]
Let us know how we can help with 1.1.0.  We're big Nukkit fans here on the ScraM team.

**Comment by Creeperface01** [comment]
@yescallop is quite experienced so maybe he could help us with update

**Comment by Snake1999** [comment]
Doing coding on 1.10. Dont worry.

**Comment by imjack** [comment]
Push what you've got maybe we can help.

**Comment by EnderKenji** [comment]
please be quick ;-; thx

**Comment by yescallop** [comment]
I've been working on it
https://github.com/yescallop/Nukkit/tree/1.1.0
Now players are able to join the server, but there are still some bugs. Trying to fix them.

**Comment by Creeperface01** [comment]
@yescallop idk why but Im getting stuck at "syncing pack with server" screen

**Comment by imjack** [comment]
@yescallop seems when walking the player thinks it's sneaking hence the shaking screen.

**Comment by ScraM-Team** [comment]
@yescallop , I will try it now with ScraM for MCPE.

**Comment by ScraM-Team** [comment]
Just got this, will try to fix:

Nukkit-1.1.0/src/main/java/cn/nukkit/level/sound/TNTPrimeSound.java:[16,36] cannot find symbol
[ERROR]   symbol:   variable EVENT_SOUND_TNT
[ERROR]   location: class cn.nukkit.network.protocol.LevelEventPacket

**Comment by antoinnneee** [comment]
I fix this error, you just need to add :
public static final int EVENT_SOUND_TNT = 1005;  
in file : /src/main/java/cn/nukkit/network/protocol/LevelEventPacket.java

**Comment by ScraM-Team** [comment]
@yescallop , it is looking great!  I can log in with 1.1.0.  The main issue I see is the "sneak shake".  The sneak button on the client is toggling on and off every half second or so, causing the screen to shake slightly up and down.  (reported by others above already).

@antoinnneee , thanks for the TNT sound tip.  I switched it to a different, valid sound (SPLASH) so that I would get some sound, even though it's not the right one.

**Comment by Goodstuff4UonYT** [comment]
@yescallop If I spawn some flaoting text particle, my skin will also spawn at the floating text particle.


<img width="768" alt="bug" src="https://cloud.githubusercontent.com/assets/19636759/26776493/a8183a12-49d9-11e7-89fd-77b248184834.png">

**Comment by DaMatrix** [comment]
I can't join with the current latest commit, I get stuck on "Locating Server".

**Comment by kvetinac97** [comment]
Now, the main things that remain to fix is touchEvent (BlockPickRequestPacket?) and inventory (add the missing slot)

**Comment by Goodstuff4UonYT** [comment]
@yescallop Now all floating text particle are invisible or don't spawn.

**Comment by boybook** [comment]
@Goodstuff4UonYT Fixed in https://github.com/yescallop/Nukkit/commit/e97811b3a957d07d6473c33e00c749c24209211f

**Comment by Snake1999** [comment]
@Goodstuff4UonYT Do NOT use float text particle anymore (it's deprecated). Use entity instead

**Comment by boybook** [comment]
@Snake1999 Not 1.10, it's 1.1.0!!

**Comment by Snake1999** [comment]
We've found a packet named `BlockPickRequestPacket`. It might be used for *selecting a block with the middle button on your mouse in creative mode (what is convenient for building in PC and Win 10 editions)*. Studying deeper on it :)

**Comment by imjack** [comment]
Fairly sure that packet is sent if the item has tag nbt like a chest, not just when you select an item.

**Comment by TDS200** [comment]
How much is left to be done and is there anything I can do to help?

**Comment by boybook** [comment]
Inventory bugs are fixed in https://github.com/yescallop/Nukkit/commit/a3c098eaba23a6dfe9d5d6502a82275ae2b839fc
If has any other bugs, please tell me. (*ﾟ∀ﾟ*)

**Comment by Goodstuff4UonYT** [comment]
@boybook 
Very nice work, I will test the floating text particle later! 
If I go/teleport to an another world, the chunks wont load and I can only see the old chunks from the old world. 
Tested with win10 and after this commit https://github.com/yescallop/Nukkit/commit/f874f366909d1c0f95119dd55805e1e57dba617f

**Comment by imjack** [comment]
Crafting seems to be broken using the latest @yescallop Build

**Comment by boybook** [comment]
Something new ^_^
https://github.com/yescallop/Nukkit/commit/ed643e70cddb1a83a32c5b0a8a29a2477982ce34

**Comment by imjack** [comment]
Any idea if enchanting table is sending the correct Id when opening?

**Comment by DaMatrix** [comment]
did it do that before? I've never been able to get the current opened GUI not be null, even on older versions of mcpe

**Comment by Creeperface01** [comment]
Idk how it is in 1.1 but in prebious versions it sent a right id but MCPE didnt send ContainerSetSlotPacket so we could not handle enchanting

**Comment by JackboyPlay** [comment]
Question btw how to compile it correctly, if I download it, uzip it, go into the folder, type mvn package and try to start the jar with "java -jar nukkit...jar" it wont start.

**Comment by TDS200** [comment]
Yes, someone please explain how to compile the nukkit.jar. Or better yet, push the .jar file out with the commits. That way we can test the new changes and find any bugs.

**Comment by xxFlare** [comment]
@TDS200 That is pretty basic java stuff you can do with any good IDE..

**Comment by pixelzee** [comment]
Get a blank folder, make sure you have git installed.  
Right click inside that folder and click "Git BASH here".
Type the command : "git clone https://github.com/yescallop/Nukkit -b 1.1.0"
Hit enter and wait for it to finish.  Then exit out of the terminal and open up the newly created folder named "Nukkit".
Again right click inside the folder and click "Git BASH here".
Type the command : "git submodule update --init"
Hit enter, then using maven type "mvn clean" then "mvn package".
The jar files will be located in the 'target' inside the directory of Nukkit

**Comment by TDS200** [comment]
@pixelzee Thank you! @xxFlare Most of my Java experience comes from developing Android apps and compiling .apk files. So, building .jars and setting up the IDE for plugin development is still pretty new to me. But, I am slowly but surely getting the hang of it. 👍

**Comment by JackboyPlay** [comment]
@pixelzee This is what happens if I execute the file:
root@JackCraft:~# cd PE-Netzwerk/
root@JackCraft:~/PE-Netzwerk# cd Vorlage/
root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
18:28:01 [INFO] Welcome! Please choose a language first!
root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
18:28:04 [INFO] Welcome! Please choose a language first!
18:28:04 [ALERT] java.lang.NullPointerException
        at java.io.Reader.<init>(Reader.java:78)
        at java.io.InputStreamReader.<init>(InputStreamReader.java:113)
        at cn.nukkit.utils.Utils.readFile(Utils.java:62)
        at cn.nukkit.Server.<init>(Server.java:222)
        at cn.nukkit.Nukkit.main(Nukkit.java:68)

18:28:04 [INFO] Stopping other threads
> root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
18:28:06 [INFO] Welcome! Please choose a language first!
> root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
18:28:07 [INFO] Welcome! Please choose a language first!
18:28:07 [ALERT] java.lang.NullPointerException
        at java.io.Reader.<init>(Reader.java:78)
        at java.io.InputStreamReader.<init>(InputStreamReader.java:113)
        at cn.nukkit.utils.Utils.readFile(Utils.java:62)
        at cn.nukkit.Server.<init>(Server.java:222)
        at cn.nukkit.Nukkit.main(Nukkit.java:68)

root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
18:28:09 [INFO] Welcome! Please choose a language first!
18:28:09 [ALERT] java.lang.NullPointerException
        at java.io.Reader.<init>(Reader.java:78)
        at java.io.InputStreamReader.<init>(InputStreamReader.java:113)
        at cn.nukkit.utils.Utils.readFile(Utils.java:62)
        at cn.nukkit.Server.<init>(Server.java:222)
        at cn.nukkit.Nukkit.main(Nukkit.java:68)

root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar --language=de
18:28:17 [INFO] Welcome! Please choose a language first!
root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar --language=de
18:28:20 [INFO] Welcome! Please choose a language first!
18:28:20 [ALERT] java.lang.NullPointerException
        at java.io.Reader.<init>(Reader.java:78)
        at java.io.InputStreamReader.<init>(InputStreamReader.java:113)
        at cn.nukkit.utils.Utils.readFile(Utils.java:62)
        at cn.nukkit.Server.<init>(Server.java:222)
        at cn.nukkit.Nukkit.main(Nukkit.java:68)

root@JackCraft:~/PE-Netzwerk/Vorlage# java -jar nukkit-1.0-SNAPSHOT.jar
18:28:23 [INFO] Welcome! Please choose a language first!
root@JackCraft:~/PE-Netzwerk/Vorlage#

**Comment by dcmwai** [comment]
Did we have any CI Build on this?

**Comment by DaMatrix** [comment]
nope, you'll have to built it yourself

**Comment by ZizzyZizzy** [comment]
@pixelzee 
Thanks for that. I've been downloading the lang files with wget..wasn't aware of the **submodule** command!

**Comment by ZizzyZizzy** [comment]
Compiled, tested and looking good so far! You guys rock!

**Comment by pixelzee** [comment]
no worries @ZizzyZizzy just remember to keep us update with any bugs you find

**Comment by jonnypeace** [comment]
Hello, firstly thanks very much for this, my daughter has been enjoying the minecraft server. :)

The 1.1.0 version flashed up some alerts, although i'm not sure what they are. It hasn't stopped us being able to use the server.

 [ALERT] java.lang.NullPointerException
	at cn.nukkit.Player.handleDataPacket(Player.java:3413)
	at cn.nukkit.network.Network.processPackets(Network.java:180)
	at cn.nukkit.network.Network.processBatch(Network.java:151)
	at cn.nukkit.Player.handleDataPacket(Player.java:1920)
	at cn.nukkit.network.RakNetInterface.handleEncapsulated(RakNetInterface.java:157)
	at cn.nukkit.raknet.server.ServerHandler.handlePacket(ServerHandler.java:123)
	at cn.nukkit.network.RakNetInterface.process(RakNetInterface.java:66)
	at cn.nukkit.network.Network.processInterfaces(Network.java:75)
	at cn.nukkit.Server.tick(Server.java:991)
	at cn.nukkit.Server.tickProcessor(Server.java:789)
	at cn.nukkit.Server.start(Server.java:768)
	at cn.nukkit.Server.<init>(Server.java:459)
	at cn.nukkit.Nukkit.main(Nukkit.java:68)

**Comment by Snake1999** [comment]
Fixed a huge bug: block breaking time. Now with the Nukkit server block breaking time can be as accurate as client, and seems more *fluent* in player's view.
https://github.com/yescallop/Nukkit/commit/32c3a4cbb2ce699d31b3a29dd7e2e1e9b3e8b287

**Comment by PetteriM1** [comment]
When the jar file comes?

**Comment by antoinnneee** [comment]
I've compille Nukkit for you PetteriM1
https://www.dropbox.com/s/3qlvgf9ttr4jz5k/nukkit-1.1.SNAPSHOT.jar?dl=0

**Comment by EnderKenji** [comment]
Thunder won't stop when its raining.
When I uses WE, I can't use pos2(right click/touch block) using the wooden axe.

**Comment by Snake1999** [comment]
@EnderKenji send a separate issue to your PLUGIN AUTHOR.

**Comment by Goodstuff4UonYT** [comment]
My win10 minecraft version crashs so oft if I connect to the server :c

**Comment by PetteriM1** [comment]
Restart the server work for me, but it happens some times.

**Comment by lsw46** [comment]
+1

**Comment by YaBoiBalAnce** [comment]
https://github.com/Nukkit/Nukkit/issues/1749

**Comment by Goodstuff4UonYT** [comment]
I cant join the server a second time, should I restart the server every time before a player join? :D

**Comment by Goodstuff4UonYT** [comment]
@boybook Your commit to the ClientChainData was amazing! It's very nice to get information like OS, language and device. Is there any possibility to get other information like if the player is using blocklauncher/toolbox or have installed some mods?

**Comment by Snake1999** [comment]
@Goodstuff4UonYT No possibility. Cheat mod authors and launcher authors prevent you from getting their information.

**Comment by jrodr10** [comment]
hmm think they would sell me the code for $200? lol @Snake1999

**Comment by Goodstuff4UonYT** [comment]
@Snake1999 okey, thanks.

**Comment by Tee7even** [comment]
So how it goes?

**Comment by Snake1999** [comment]
Not yet. Ramping down bugs :)

**Comment by Goodstuff4UonYT** [comment]
@boybook you can't cancel the InventoryClickEvent? Is there any other possibility to avoid that the win 10 player can pick a item and drag it around?

**Comment by boybook** [comment]
Sorry, I have not studied this method, so now temporarily for the TODO state.

**Comment by MasterlazorX** [comment]
@Snake1999 Block breaking time issue still exists, tested on local.

**Comment by Snake1999** [comment]
@MasterlazorX The time calculating is CORRECT (matches what minecraft wiki says), but the client seems strange receiving "correct" information. Having further checks in it.

**Comment by boybook** [comment]
I did a test for this, as long as do Thread.sleep(pause) a short time after sending StartGamePacket, the game will not crash again.
Please personally test to determine the correctness of this discovery.

**Comment by Alex0Plays** [comment]
Left click doesen't work.
![image](https://user-images.githubusercontent.com/9166661/27289902-32e27f46-5514-11e7-8145-048c4b1af26d.png)
![image](https://user-images.githubusercontent.com/9166661/27289933-56f6376a-5514-11e7-8ba1-27f5f4b40920.png)

**Comment by Creeperface01** [comment]
is it PE or windows 10?

**Comment by Alex0Plays** [comment]
Windows 10, but same problem for pe too

**Comment by Creeperface01** [comment]
Its probably start break

**Comment by Alex0Plays** [comment]
Block's doesn't appear correct after changing with plugins like worldedit. It's normal only after rejoin.

![image](https://user-images.githubusercontent.com/9166661/27358530-ce18f7fe-561f-11e7-8181-13864e1abcc0.png)

**Comment by chadsnet** [comment]
That last 1.1.0 .jar work great for getting my mobile up on the server, but still can't get the Windows 10 client to join.  What drives me crazy is how am I able to join random servers out there that are older versions.  Hell I joined one that was 1.0.0.  (this is my first day with your software and minecraft in general)

**Comment by optimusprime338** [comment]
hey, you can join server by going to a pocketmine server and directly get transfferd to the server!!

**Comment by EnderKenji** [comment]
The chunk bug still a thing, just not happens too often,

Bugs I knw:
- When raining, thunder won't stop.
- When I do /time stop , the time wont stop. 

Using latest build.

**Comment by kpwnh** [comment]
![image](https://user-images.githubusercontent.com/16330761/27555183-9ad07542-5a76-11e7-9446-1d62bc430661.png)
Hello! I'm having a bug that honestly stresses me out:

 - OPs can't use any commands.
 - Chunks not loading AND screen is shaking
 - Client crashes constantly
 - No movement at all, stuck shaking in a point.

I'm using yescallop/Nukkit@46a28a0 at the moment, I already tested : 1033c04 also and the same issues still happen.

The server is running Ubuntu at the moment, already tested with CentOS.

Client runs Windows 10, no way to test in PE because I don't have any PE device.

Thanks :D

**Comment by optimusprime338** [comment]
wonder I m using it with no problem!!!

**Comment by kpwnh** [comment]
Found a way to bypass this error, when there's no worlds and the generator is set to make DEFAULT world, it causes this, when it's to FLAT, it gets fixed :D

**Comment by averagesecurityguy** [comment]
I've been using the 1.1.0 version from @yescallop for the last day. I can't make a furnace, which is a known issue and blocks don't want to break the first time. Other than that things seem to be working well. Thanks for the software, my kids really appreciate having their own server.

**Comment by Snake1999** [comment]
As 1.1 now merged to master, all problems and bugs, please create issue in nukkit/nukkit.

**Comment by kpwnh** [comment]
Thanks
